torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -444,6 +472,21 @@ struct GraphFuser {
     if (!shouldFuse) {
       return at::nullopt;
     }
+    if (producer->node()->kind() == aten::_grad_sum_to_size &&
+	consumer->kind() == prim::FusionGroup) {
```

Comment:
Can we assert that `consumer` is a fusion group? If I understand correctly this is the only case possible by design, but having a check like this would mean that this case will simply get skipped if it wasn't the case.

Reply:
I think the case where the consumer isn't a fusion group, i.e. a new fusion group will be started with a `_grad_sum_to_size` for the output, is legitimate, and then we would have the empty set to check.
The main alternative I see would be to refuse to fuse here, but that would only result in us having two _grad_sum_to_sizes in a row (one pushed from inside the fusion group and one that we didn't fuse because it was at the output) during execution.